apps/opik-backend/src/main/java/com/comet/opik/domain/LlmProviderApiKeyService.java

```diff
@@ -123,12 +124,30 @@ public void updateApiKey(@NonNull UUID id, @NonNull ProviderApiKeyUpdate provide
         });
     }
 
+    @Override
+    public void delete(UUID id, String workspaceId) {
+        template.inTransaction(WRITE, handle -> {
+
+            var repository = handle.attach(LlmProviderApiKeyDAO.class);
+
+            int deleted = repository.delete(id, workspaceId);
+
+            // Check that api key with such id was actually deleted
+            if (deleted == 0) {
+                throw notFoundError();
+            }
```

Comment:
If you take a look at the rest of deletion endpoints, they never return a 404 if something isn't found. The reason is idempotency, specially for scenarios where there's a client retry (e.g: due to transient network connectivity issues). 

The downside is that users won't get a significative error code for un-existing key ids, but that's better than getting an exception after a retry.

This is a blocker.

Reply:
Updated